pandas/core/indexes/base.py

```diff
@@ -792,14 +808,19 @@ def _cleanup(self) -> None:
     @cache_readonly
     def _engine(
         self,
-    ) -> libindex.IndexEngine | libindex.ExtensionEngine:
+    ) -> libindex.IndexEngine | libindex.ExtensionEngine | libindex.MaskedIndexEngine:
         # For base class (object dtype) we get ObjectEngine
         target_values = self._get_engine_target()
-        if (
-            isinstance(target_values, ExtensionArray)
-            and self._engine_type is libindex.ObjectEngine
-        ):
-            return libindex.ExtensionEngine(target_values)
+        if isinstance(target_values, ExtensionArray):
+            from pandas.core.arrays import BaseMaskedArray
```

Comment:
cant do this up top?

Reply:
hmm that sucks.  not for this PR but i bet either Categorical or IntervalArray are importing something from indexes and it would be reasonable to make the runtime imports happen there